lib/multi.c

```diff
@@ -2019,6 +2019,14 @@ static CURLMcode multi_runsingle(struct Curl_multi *multi,
       break;
 
     case MSTATE_DO:
+      /* At this point this connection is definitely connected, so trigger
+         the progress callback if it's been configured. */
+      Curl_set_in_callback(data, true);
+      if(data->set.fxferinfo != NULL) {
+        data->set.fxferinfo(data->set.progress_client, 0, 0, 0, 0);
+      }
+      Curl_set_in_callback(data, false);
```

Comment:
1. Why not check the func pointer *before* the `Curl_set_in_callback()` call?
2. Skip the `!= NULL` from the comparison, we don't use those in non-NULL comparisons everywhere
3. You don't handle the return code from the callback